casepro/contacts/models.py

```diff
@@ -184,6 +184,17 @@ def get_or_create(cls, org, uuid, name=None):
             return contact
 
     @classmethod
+    def get_or_create_from_urn(cls, org, urn, name=None):
+        """
+        Gets an existing contact or creates a stub contact. Used when opening a case without an initial message
+        """
+        contact = cls.objects.filter(urns__contains=[urn]).first()
```

Comment:
This needs to perform some sort normalization on the URN e.g. tel:+27123456789 should match tel:+27-12345-6789.

If it's helpful you can take a look at the URN class in RapidPro https://github.com/rapidpro/rapidpro/blob/master/temba/contacts/models.py#L64